packages/UI/src/components/PDFSignerViewComponent.vue

```diff
@@ -0,0 +1,171 @@
+          {{ loadingAcceptBtn ? "" : "Accept" }}
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import axios from "axios";
+import { ThreefoldWalletConnectorApi } from "tf-wallet-connector-api";
+import { onMounted, ref } from "vue";
+import { createLoadingTask, VuePdf } from "vue3-pdfjs";
+import { type VuePdfPropsType } from "vue3-pdfjs/components/vue-pdf/vue-pdf-props"; // Prop type definitions can also be imported
+
+import { KeypairType, sign } from "@/utils/sign";
+import { AlertType, type PDFPostData } from "@/utils/types";
+
+import CustomAlertComponent from "./CustomAlertComponent.vue";
+import LoadingSpinnerComponent from "./LoadingSpinnerComponent.vue";
+
+export default {
+  name: "PDFSignerViewComponent",
+  props: ["pdfurl", "dest"],
+  components: {
+    VuePdf,
+    CustomAlertComponent,
+    LoadingSpinnerComponent,
+  },
+  setup(props) {
+    const alertType = AlertType;
+    const numOfPages = ref(0);
+    const loadingPdf = ref(false);
+    const isError = ref(false);
+    const errorMessage = ref("");
+    const pdfData = ref<string>("");
+    const isAcceptDisabled = ref(true);
+    const loadingAcceptBtn = ref(false);
+    const isInstalled = ref();
+    const hasAccess = ref();
+
+    const pdfSrc = ref<VuePdfPropsType["src"]>(props.pdfurl);
+    console.log("pdfSrc: ", pdfSrc);
+
+    onMounted(async () => {
+      loadingPdf.value = true;
+      try {
+        if (!props.pdfurl) {
+          isError.value = true;
+          errorMessage.value = "An error occurred while loading the PDF: The property `pdfurl` should be provided.";
+          return;
+        }
+
+        isInstalled.value = await ThreefoldWalletConnectorApi.isInstalled();
+        hasAccess.value = await ThreefoldWalletConnectorApi.hasAccess();
+
+        const loadingTask = createLoadingTask(props.pdfurl);
+        const pdf = await loadingTask.promise;
+        const data = await pdf.getData();
+        pdfData.value = data.toString();
+        numOfPages.value = pdf.numPages;
+      } catch (error: any) {
+        isError.value = true;
+        errorMessage.value =
+          `An error occurred while loading the PDF: ${error.message}` || "An error occurred while loading the PDF.";
+        console.error(error);
+      } finally {
+        loadingPdf.value = false; // Set loading to false when PDF is loaded or errored
+      }
+    });
+
+    const accept = async () => {
+      isAcceptDisabled.value = loadingAcceptBtn.value = true;
+      if (pdfData.value) {
+        if (!hasAccess.value) {
+          hasAccess.value = await ThreefoldWalletConnectorApi.requestAccess();
+        }
+        const account = await ThreefoldWalletConnectorApi.selectDecryptedAccount();
+        const data = await sign(pdfData.value, account?.mnemonic ?? "", KeypairType.ed25519);
+        const requestBody: PDFPostData = {
+          pdfUrl: props.pdfurl,
+          pubkey: account?.address || "",
+          signature: data,
```

Comment:
please check the return data from the sign method. it contains the address and the signature